apps/dashboard/src/@/api/analytics.ts

```
@@ -424,3 +428,95 @@ export async function getEngineCloudMethodUsage(
   const json = await res.json();
   return json.data as EngineCloudStats[];
 }
+
+export async function getInsightChainUsage(
+  params: AnalyticsQueryParams,
+): Promise<InsightChainStats[]> {
+  const searchParams = buildSearchParams(params);
+  const res = await fetchAnalytics(
+    `v2/insight/usage/by-chain?${searchParams.toString()}`,
+    {
+      method: "GET",
+    },
+  );
+
+  if (res?.status !== 200) {
+    const reason = await res?.text();
+    console.error(
+      `Failed to fetch Insight chain usage: ${res?.status} - ${res.statusText} - ${reason}`,
```

This is silently failing. we won't know if there's something wrong or not in the frontend if the request is failing or just returning empty data

please return `{ data: [] } | { errorMessage: ''}` instead and show the error in frontend

---

Done